src/sentry/api/endpoints/organization_integration_request.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+
+from django.core.urlresolvers import reverse
+
+from rest_framework.response import Response
+
+from sentry import options
+from sentry.api.bases.organization_integrations import (
+    OrganizationIntegrationBaseEndpoint,
+    OrganizationIntegrationsPermission,
+)
+from sentry.models import Integration, OrganizationMember
+from sentry.utils.email import MessageBuilder
+from sentry.utils.http import absolute_uri
+
+
+class OrganizationIntegrationRequestEndpoint(OrganizationIntegrationBaseEndpoint):
+    permission_classes = (OrganizationIntegrationsPermission,)
+
+    def post(self, request, organization, integration_id):
+        """
+        Email the default organization owner asking to install an integration.
+        ``````````````````````````````````````````````````````````````````````
+
+        TODO 0.0 descibe why this exists.
+
+        :param string message: Optional message from the requester to the owner.
+        """
+
+        # If for some reason the user had permissions all along, silently fail.
+        requester = request.user
+        if requester.id in map(lambda user: user.id, organization.get_owners()):
```

Comment:
@mgaeta I don't think we necessarily have to handle this case. Your call if you want to keep it but I think it's totally fine to email the owners even if an owner is suggesting this since the UI should prevent this from happening.